util.rb

```diff
@@ -4,58 +4,48 @@
 # IAM - a resource usage metric collection and reporting system
 class Util
   def self.max_value(data)
-    if data.empty?
-      return 0
-    end
+    return 0 if data.empty?
     max = data[0][:value]
     data.each do |item|
       compare_max = item[:value]
-      if max < compare_max
-        max = compare_max
-      end
+      max = compare_max if max < compare_max
     end
-    return max
+    max
   end
 
   def self.min_value(data)
     return 0 if data.empty?
     min = data[0][:value]
     data.each do |item|
       compare_min = item[:value]
-      if min > compare_min
-        min = compare_min
-      end
+      min = compare_min if min > compare_min
     end
-    return min
+    min
   end
 
   def self.average_value(data)
-    if data.empty?
-      return 0
-    end
+    return 0 if data.empty?
     average = data[0][:value]
     count = 1
     data.slice(1..-1).each do |item|
       average += item[:value]
       count += 1
     end
     average /= count
-    return average
+    average
   end
 
-  def self.is_positive_integer_value(data)
+  def self.positive_integer_value?(data)
     flag = true
     i = 0
     while i < data.length && flag != false
       if data[i][:value].is_a? Integer
-        if data[i][:value] < 0
-          flag = false
-        end
+        flag = false if data[i][:value] < 0
       else
         flag = false
       end
       i += 1
     end
-    return flag
+    flag
```

Comment:
While it's true that procedural languages work well with a single entry and a single exit, that's not really the case with Ruby.  Ruby's idiomatic style is pretty different from C, Perl, Python, Go and darn near every other language I can think of right now. :-)  https://github.com/bbatsov/ruby-style-guide is a reasonable description of idiomatic Ruby style in the wild, and it also does a great job of showing the potential dangers in using return as you might do in other languages -- something I didn't know until just now!